src/backend/utils/misc/guc_gp.c

```diff
@@ -5472,3 +5475,17 @@ DispatchSyncPGVariable(struct config_generic * gconfig)
 
 	CdbDispatchSetCommand(buffer.data, false);
 }
+
+static void
+assign_gp_interconnect_type(int newval, void *extra)
+{
+	Gp_interconnect_type = newval;
+
+	/*
+	 * "interconnect.so" is not currently loaded if CurrentIPCLayerImplNum == 0.
+	 */
+	if (CurrentIPCLayerImplNum != 0)
+	{
+		SetCurrentMotionIPCLayer(Gp_interconnect_type);
```

Comment:
I wonder how to release prev resources ?

Reply:
The description in commit is ambiguous, I will clear it.